src/ensembl/src/shared/components/layout/StandardAppLayout.tsx

```diff
@@ -0,0 +1,152 @@
+import { ReactComponent as CloseIcon } from 'static/img/shared/close.svg';
+
+import styles from './StandardAppLayout.scss';
+
+enum SidebarModeToggleAction {
+  OPEN = 'open',
+  CLOSE = 'close'
+}
+
+type SidebarModeToggleProps = {
+  onClick: () => void;
+  showAction: SidebarModeToggleAction;
+};
+
+type SidebarNavigationProps = {
+  links: Array<{
+    label: string;
+    isActive: boolean;
+  }>;
+  onChange: (index: number) => void;
+};
+
+type StandardAppLayoutProps = {
+  mainContent: ReactNode;
+  sidebarContent: ReactNode;
+  sidebarToolstripContent?: ReactNode;
+  sidebarNavigation: SidebarNavigationProps;
+  topbarContent: ReactNode;
+  isSidebarOpen: boolean;
+  onSidebarToggle: () => void;
+  isDrawerOpen: boolean;
+  drawerContent?: ReactNode;
+  onDrawerClose: () => void;
+};
+
+const StandardAppLayout = (props: StandardAppLayoutProps) => {
+  const mainClassnames = classNames(
+    styles.main,
+    { [styles.mainDefault]: props.isSidebarOpen },
+    { [styles.mainFullWidth]: !props.isSidebarOpen }
+  );
+
+  const sidebarWrapperClassnames = classNames(
+    styles.sideBarWrapper,
+    { [styles.sideBarWrapperOpen]: props.isSidebarOpen },
+    { [styles.sideBarWrapperClosed]: !props.isSidebarOpen },
+    {
+      [styles.sideBarWrapperDrawerOpen]:
+        'isDrawerOpen' in props ? props.isDrawerOpen : false
+    }
+  );
+
+  return (
+    <div className={styles.standardAppLayout}>
+      <div className={styles.topBar}>
+        {props.topbarContent}
+        <SidebarTabs {...props} />
+      </div>
+      <div className={styles.mainWrapper}>
+        <div className={mainClassnames}>{props.mainContent}</div>
+        <div className={sidebarWrapperClassnames}>
+          {props.isDrawerOpen && <DrawerWindow onClick={props.onDrawerClose} />}
+          <div className={styles.sideBarToolstrip}>
+            <SidebarModeToggle
+              onClick={
+                props.isDrawerOpen ? props.onDrawerClose : props.onSidebarToggle
+              }
+              showAction={
+                props.isSidebarOpen
+                  ? SidebarModeToggleAction.CLOSE
+                  : SidebarModeToggleAction.OPEN
+              }
+            />
+            <div className={styles.sideBarToolstripContent}>
+              {props.sidebarToolstripContent}
+            </div>
+          </div>
+          <div className={styles.sideBar}>{props.sidebarContent}</div>
+          <div className={styles.drawer}>
+            <CloseIcon
+              className={styles.drawerClose}
+              onClick={props.onDrawerClose}
+            />
+            {props.drawerContent || null}
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+StandardAppLayout.defaultProps = {
+  isDrawerOpen: false,
+  onDrawerClose: noop
+};
+
+const SidebarModeToggle = (props: SidebarModeToggleProps) => {
+  const chevronClasses = classNames(styles.sidebarModeToggleChevron, {
+    [styles.sidebarModeToggleChevronOpen]:
```

Comment:
This is not defined in the styles and also how about just `toggleChevronOpen` if it is required?

Reply:
It is defined; otherwise the chevron wouldn't rotate :-)
https://github.com/Ensembl/ensembl-client/pull/220/files#diff-28418f7ce136b1f5ddf0c4bf3d65f653R104-R106

Reply:
Oh right! I didn't know `sidebarModeToggleChevron{ &Open..` can be used as `sidebarModeToggleChevronOpen`